pennylane/operation.py

```diff
@@ -373,71 +270,243 @@ def classproperty(func):
-        return str([qml.math.round(qml.math.real(d) % (2 * np.pi), 10) for d in op.data])
+    An operator can define any of the following representations:
 
-    if op.name in ("CRX", "CRY", "CRZ", "CRot"):
-        return str([qml.math.round(qml.math.real(d) % (4 * np.pi), 10) for d in op.data])
+    * Representation as a **matrix** (:meth:`.Operator.matrix`), as specified by a
+      global wire order that tells us where the wires are found on a register.
 
-    return str(op.data)
+    * Representation as a **sparse matrix** (:meth:`.Operator.sparse_matrix`). Currently, this
+      is a scipy coo matrix format.
 
+    * Representation via the **eigenvalue decomposition** specified by eigenvalues
+      (:meth:`.Operator.eigvals`) and diagonalizing gates (:meth:`.Operator.diagonalizing_gates`).
 
-class Operator(abc.ABC):
-    r"""Base class for quantum operators supported by a device.
+    * Representation as a **product of operators** (:meth:`.Operator.decomposition`).
 
-    The following class attributes must be defined for all Operators:
+    * Representation as a **linear combination of operators** (:meth:`.Operator.terms`).
 
-    * :attr:`~.Operator.num_wires`
+    Each representation method comes with a static method prefixed by ``compute_``, which
+    takes the signature ``(*parameters, **hyperparameters)`` (for numerical representations that do not need
+    to know about wire labels) or ``(*parameters, wires, **hyperparameters)``, where ``parameters``, ``wires``, and
+    ``hyperparameters`` are the respective attributes of the operator class.
 
     Args:
-        params (tuple[float, int, array]): operator parameters
-
-    Keyword Args:
-        wires (Iterable[Number, str], Number, str, Wires): Wires that the operator acts on.
+        params (tuple[tensor_like]): trainable parameters
+        wires (Iterable[Any] or Any): Wire label(s) that the operator acts on.
             If not given, args[-1] is interpreted as wires.
-        do_queue (bool): Indicates whether the operator should be
-            immediately pushed into the Operator queue.
+        do_queue (bool): indicates whether the operator should be
+            recorded when created in a tape context
+        id (str): custom label given to an operator instance,
+            can be useful for some applications where the instance has to be identified
+
+    **Example**
+
+    A custom operator can be created by inheriting from :class:`~.Operator` or one of its subclasses.
+
+    The following is an example for a custom gate that possibly flips a qubit and then rotates another qubit.
+    The custom operator defines a decomposition, which the devices can use (since it is unlikely that a device
+    knows a native implementation for ``FlipAndRotate``). It also defines an adjoint operator.
+
+    .. code-block:: python
+
+        import pennylane as qml
+
+
+        class FlipAndRotate(qml.operation.Operation):
```

Comment:
I wonder if it might be confusing that the example uses `Operation` and not `Operator`?

Reply:
I explained it in the text!